File: dynamic_listings/javascript/default/procedural/createrandomizedweapon.js
function createRandomizedWeapon(){
    // The "class" and "new" keywords are supported in ES6/ECMAScript 2015
    let wp = new Weapon();
    // Load some randomized components
    wp.loadRandomBodyFrom("weaponBodies/shotguns");
    wp.loadRandomScopeFrom("weaponScopes/shotguns");
    wp.loadRandomBarrelFrom("weaponBarrels/shotguns");
    wp.loadRandomAmmoMagazineFrom("weaponMagazines/shotguns");
    // set weapon damage to a value between 35 and 50
    wp.damage = Math.random() * 16 + 35;
    // set weapon range to a value between 13 and 18
    wp.range = Math.random() * 5 + 13;
    // set weapon spread to a value between 20 and 30
    wp.spread = Math.random() * 10 + 20;
    return wp;
}

File: dynamic_listings/python/default/procedural/createrandomizedweapon.py
from random import choice as random_choice
from random import randint
from os import listdir


def create_randomized_weapon() -> Weapon:
    # Start by getting the directory contets for each component
    bodies: list[str] = listdir("weaponBodies/shotguns")
    scopes: list[str] = listdir("weaponScopes/shotguns")
    barrels: list[str] = listdir("weaponBarrels/shotguns")
    mags: list[str] = listdir("weaponMagazines/shotguns")
    # Choose a random piece for each component
    body: Sprite = Sprite(random_choice(bodies))
    scope: Sprite = Sprite(random_choice(scopes))
    barrel: Sprite = Sprite(random_choice(barrels))
    ammo_mag: Sprite = Sprite(random_choice(mags))
    weapon: Weapon = compose_weapon(body, scope, barrel, ammo_mag)
    # Set weapon damage to a value between 35 and 50
    weapon.set_damage(randint(35, 50))
    # Set weapon range to a value between 13 and 18
    weapon.set_range(randint(13,28))
    # Set weapon spread to a value between 20 and 30
    weapon.set_spread(randint(20,30))
    return weapon

